src/lib/configuration_objects.py

```diff
@@ -20,10 +23,13 @@ def __init__(self, coding_mode, code_scheme, coded_field, fold_strategy, raw_fie
         self.include_in_theme_distribution = include_in_theme_distribution
 
 
-# TODO: Rename CodingPlan to something like DatasetConfiguration?
 class CodingPlan(object):
     def __init__(self, raw_field, coding_configurations, raw_field_fold_strategy, coda_filename=None, ws_code=None,
-                 time_field=None, run_id_field=None, icr_filename=None, id_field=None, code_imputation_function=None):
+                 time_field=None, run_id_field=None, icr_filename=None, id_field=None, code_imputation_function=None,
+                 message_id_fn=None):
+        if message_id_fn is None:
+            message_id_fn = lambda td: SHAUtils.sha_string(td[self.raw_field])
```

Comment:
why are still using the raw field here?

Reply:
I didn't want to need to explicitly set the message_id_fn function in all cases, so the default behaviour is to do what we've always done and compute the message id by sha-ing the raw message text. This lets the sms pipeline continue running unmodified.
If you think it's better not to have a default, we could set this explicitly everywhere, but it would still be set to use the raw field in the sms cases.

Reply:
makes sense let's keep it as is.